### ISSUE TITLE
Bioblitz project aggregating obs based on posted_on rather than observed_on (I think)

### ISSUE THREAD

**Issue by loarie** [inaturalist/inaturalist#803]
This bioblitz project http://www.inaturalist.org/projects/cormorant-week-jan-3-9 has the aggregator turned on and is including several obs which seem to have been added by the aggregator (project_observation.user_id=nil) that were POSTED during the bioblitz time interval but not OBSERVED during that interval...
[2558340, 2552761, 2550398, 2550088, 2550358, 2549728, 2558345, 2558637, 2558899, 2559306, 2559311, 2559328, 2559475, 2559477, 2559484, 2559493, 2559494, 2559490, 2559496, 2559514, 2559580, 2560168, 2560745, 2560775, 2563246, 2560880, 2561042, 2552091, 2561947, 2561953, 2562434, 2562490, 2563320, 2563068, 2563252] 

Is that whats going on? Can't figure out how else they were added. I didn't think we were capable of detecting obs POSTED during an interval with the current search API....

**Comment by loarie** [comment]
Not sure if this is related to the above, but here's another glitch where the user's membership stats don't reflect the one project_observation (ie says 0 in several places)
http://www.inaturalist.org/projects/cormorant-week-jan-3-9/members
<img width="671" alt="screen shot 2016-01-11 at 11 41 18 am" src="https://cloud.githubusercontent.com/assets/598026/12244375/43dd11d0-b858-11e5-9d22-017d51e57e63.png">
http://www.inaturalist.org/projects/cormorant-week-jan-3-9/contributors/scubale
<img width="988" alt="screen shot 2016-01-11 at 11 41 12 am" src="https://cloud.githubusercontent.com/assets/598026/12244377/45362058-b858-11e5-900b-6575fa858b22.png">

**Comment by loarie** [comment]
Any insight into this? Here's another way to see the discrepancy:
http://staging.inaturalist.org/observations?d1=2016-01-03T00:00:00-08:00&d2=2016-01-10T00:00:00-08:00&taxon_id=71268
vs.
http://staging.inaturalist.org/observations?project_id=cormorant-week-jan-3-9

**Comment by pleary** [comment]
There were some observations in the project which were observed outside the BioBlitz window, but they had been added by users before that privilege was revoked. Scott said there was at least one he thought was deleted and came back, but I can't confirm that. I tried on Staging and using the "Remove" option from the project list on observations show did seem to work for me.

Not sure exactly what happened with the counts. There are delayed jobs to update a cached observation count on ProjectUser, so it is possible we are failing to call that method during some create/update/delete. I manually reset the counts with:

```Project.find(5935).project_users.each{ |pu| ProjectUser::update_observations_counter_cache_from_project_and_user(5935, pu.user_id) }``

The only users that were wrong were scubale and tiwane which first reported 0, then reported 1 after updating the cached count. Ultimately not much was resolved here. If we can get more data maybe we can track down the operation that's causing counts to get out of sync.
